.github/workflows/semver-checks.yml

```diff
@@ -34,7 +34,7 @@ jobs:
       - name: "Install cargo-binstall"
         uses: cargo-bins/cargo-binstall@main
       - name: "Binstall cargo-semver-checks"
-        run: cargo binstall cargo-semver-checks --no-confirm
+        run: cargo binstall cargo-semver-checks@0.32.0 --no-confirm
```

Comment:
The version needs to be pulled from a file that's updated every Friday together with Rust version.

Reply:
Maybe we should upgrade the flat nightly-version text file to a JSON file that contains the nightly version, stable version for cargo-semver-checks, version of cargo-semver-checks, ... and maybe also the stable/beta versions for the rest of CI (which aren't currently pinned because in theory they should never break us).

Reply:
Yes that's what I thought so. But @apoelstra said to mimic how we used to pin `cargo-public-api`.

What if we leave this as it is now, and move the dynamic pinning of `cargo-semver-checks` to an issue and add in a new PR?

Reply:
I was mistaken. With `cargo-public-api` we didn't really have a plan to update the pin (actually I think we intended to just never update it because updates basically only ever broke us). Sorry about that.

Reply:
I would prefer not want to deal with json in scripts since it makes scripts less readable IMO. If you want it in a single file perhaps just use the shell `key=value` and source it.

Reply:
That will make it difficult to parse using anything but shell.